### ISSUE TITLE
how to keep getting notified on completed uploads after the component is destroyed?

### ISSUE THREAD

**Issue by mrmkrs** [bleenco/ngx-uploader#459]
Short story: I would like to receive the 'done' event after the component gets destroyed.

Background: In order to achieve the perception of a 'fast responding' application, I would like to completely async all the file uploads. 

The scenario: 
1. User opens some form (lets say ArticleFormComponent)
2. Select some big files
3. Click save
4. The form data without the files is submitted to the API
5. Upon successful request, all files are uploaded

Between steps 4 and 5, I would like to close the form (component) and show the submitted data in an ArticleDetailsComponent. In the meantime, the browser should still be uploading. For every uploaded file, I would like to have an event (done) so I can show the file in the ArticleDetailsComponent. 

So after clicking submit: The user should see the form getting hidden, redirected to the just submitted article. And as long as the user is viewing the ArticleDetailsComponent, for every completed fileupload in the background the ArticleDetailsComponent is updated showing the completed file.

The problem is that I can never receive the done event since the ArticleFormComponent was destroyed long before any upload got completed, so as a work around I have to keep it open while there are pending uploads.

**Comment by jkuri** [comment]
why dont you use module inside some global service instead of component? that would solve your issue. demo example is just an sample and not actual best use practice, thats why its triggered from component.

**Comment by mrmkrs** [comment]
Yes I understand, but I don't think that fixes the problem. I have setup a demo here: https://stackblitz.com/edit/angular-c1mt73 

There is a separate UploaderComponent that renders the <file> input. A parent component can talk to the uploader through the UploaderService which acts like a message bus.

If you upload a big file, the 'done' event is never printed in the console unless you comment this.showUploader = false; in the AppComponent
